js/runtime.js

```diff
@@ -737,15 +737,15 @@ function Fay$$writeRef(ref,x){
 }
 
 // Get the value from the ref.
-function Fay$$readRef(ref,x){
+function Fay$$readRef(ref){
   return ref.value;
 }
 
 /*******************************************************************************
  * Dates.
  */
 function Fay$$date(str){
-  return window.Date.parse(str);
```

Comment:
Is this change because there's no typescript extern for `window`? I can't recall why we used `window` in the first place, but it may have to do with name shadowing...

Reply:
You can get the global object without naming it by doing `(function () { return this; }).apply()`, does that work in TypeScript?

But no need to revert your change unless anyone can remember why `window` was there in the first place...

Reply:
> Is this change because there's no typescript extern for window?

Yes. And [ECMAScript 2015](http://www.ecma-international.org/ecma-262/6.0/) does not need a reference of window-object.